lib/containers/github-header-container.js

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {QueryRenderer, graphql} from 'react-relay';
+
+import {incrementCounter} from '../reporter-proxy';
+import {EndpointPropType, RemoteSetPropType} from '../prop-types';
+import RelayNetworkLayerManager from '../relay-network-layer-manager';
+import {UNAUTHENTICATED, INSUFFICIENT} from '../shared/keytar-strategy';
+import ObserveModel from '../views/observe-model';
+import QueryErrorView from '../views/query-error-view';
+import GithubHeaderView from '../views/github-header-view';
+
+export default class GithubHeaderContainer extends React.Component {
+  static propTypes = {
+    // Connection
+    loginModel: PropTypes.object.isRequired,
+    endpoint: EndpointPropType.isRequired,
+
+    avatarUrl: PropTypes.string,
+    remotes: RemoteSetPropType.isRequired,
+    currentRemoteName: PropTypes.string.isRequired,
+
+    handleRemoteSelect: PropTypes.func.isRequired,
+  }
+
+  fetchToken = loginModel => {
```

Comment:
you should note that this is duplicate code from `remote-controller.js`, and I believe that this duplicate logic should be separated to a component that handles the login/endpoint logic for GraphQL however since it is just token verification functions and one component I was unsure what to call it some ideas I had were some combination of `github-` + (`login-` or `token-`) + (`controller` or `container`)